src/storage/page_buffer.c

```diff
@@ -1975,15 +1976,32 @@ pgbuf_fix_release (THREAD_ENTRY * thread_p, const VPID * vpid, PAGE_FETCH_MODE f
     }
   else
     {
-      bufptr = pgbuf_claim_bcb_for_fix (thread_p, vpid, fetch_mode, hash_anchor, &perf, &retry);
+      // TODO: more accurate comments
+      /* In this case, the caller is holding only hash_anchor->hash_mutex. The hash_anchor->hash_mutex is to be
+       * released in pgbuf_lock_page (). */
+      const PGBUF_LOCK_MODE lock_page_res = pgbuf_lock_page (thread_p, vpid, fetch_mode, hash_anchor, &perf);
+      if (lock_page_res == PGBUF_LOCK_UNKNOWN)
+	{
+	  ASSERT_ERROR ();
+	  return nullptr;
+	}
+      else if (lock_page_res == PGBUF_LOCK_WAITER)
+	{
+	  goto try_again;
+	}
+      assert (lock_page_res == PGBUF_LOCK_HOLDER);
+
+      bufptr = pgbuf_claim_bcb_for_fix (thread_p, vpid, fetch_mode);
       if (bufptr == NULL)
 	{
-	  if (retry)
-	    {
-	      retry = false;
-	      goto try_again;
-	    }
 	  ASSERT_ERROR ();
+	  // TODO: more explanatory comments
+	  /*
+	   * Now, caller is not holding any mutex.
+	   * the last argument of pgbuf_unlock_page () is true that
+	   * means hash_mutex must be held before unlocking page.
+	   */
```

Comment:
```suggestion
	  // No mutex is held. The last argument of pgbuf_unlock_page () is true  because hash_mutex must be locked before unlocking page.
```